chroma_api/stratagem.py

```diff
@@ -27,71 +27,158 @@
+    elif ManagedFilesystem.objects.get(id=fs_id).state != "available":
+        return {"code": "filesystem_unavailable", "message": "Filesystem {} is unavailable.".format(fs_identifier)}
+
+
+def get_target_mount_ids(fs_id, bundle):
+    # At least Mdt 0 should be mounted, or stratagem cannot run.
+    target_mount_ids = (
+        ManagedMdt.objects.filter(filesystem_id=fs_id, active_mount_id__isnull=False)
+        .values_list("active_mount_id", flat=True)
+        .distinct()
+    )
+
+    return target_mount_ids
+
+
+def validate_target_mount(bundle, result):
+    if result:
+        return result
+
+    (r, fs_id) = get_fs_id(bundle)
+    if isinstance(r, dict):
+        return r
+
+    # At least Mdt 0 should be mounted, or stratagem cannot run.
+    target_mount_ids = get_target_mount_ids(fs_id, bundle)
+    mdt0 = ManagedMdt.objects.filter(filesystem_id=fs_id, name__contains="MDT0000").first()
+
+    if mdt0 is None:
+        return {"code": "mdt0_not_found", "message": "MDT0 could not be found."}
+
+    if mdt0.active_mount_id not in target_mount_ids:
+        return {"code": "mdt0_not_mounted", "message": "MDT0 must be mounted in order to run stratagem."}
+
+
+def validate_mdt_profile(bundle, result):
+    if result:
+        return result
+
+    (r, fs_id) = get_fs_id(bundle)
+    if isinstance(r, dict):
+        return r
+
+    # At least Mdt 0 should be mounted, or stratagem cannot run.
+    target_mount_ids = get_target_mount_ids(fs_id, bundle)
+
+    host_ids = ManagedTargetMount.objects.filter(id__in=target_mount_ids).values_list("host_id", flat=True).distinct()
+
+    installed_profiles = (
+        ManagedHost.objects.filter(id__in=host_ids).values_list("server_profile_id", flat=True).distinct()
+    )
+
+    if not all(map(lambda name: name == "stratagem_server", installed_profiles)):
+        return {
+            "code": "stratagem_server_profile_not_installed",
+            "message": "'stratagem_server' profile must be installed on all MDT servers.",
+        }
+
+
+def validate_client_profile(bundle, result):
+    if result:
+        return result
+
+    if not ManagedHost.objects.filter(server_profile_id="stratagem_client").exists():
+        return {
+            "code": "stratagem_client_profile_not_installed",
+            "message": "A client must be added with the 'Stratagem Client' profile to run this command.",
+        }
+
+
+class RunStratagemValidation(Validation):
+    def is_valid(self, bundle, request=None):
+        return (
+            pipe(
+                None,
+                partial(validate_duration, bundle),
+                partial(validate_filesystem, bundle),
+                partial(validate_target_mount, bundle),
+                partial(validate_mdt_profile, bundle),
+                partial(validate_client_profile, bundle),
+            )
+            or {}
+        )
 
 
 class StratagemConfigurationValidation(RunStratagemValidation):
```

Comment:
How is this error being accounted for on the Rust side:

`return {"__all__": "Required fields are missing: {}".format(", ".join(difference))}`

Reply:
Good catch. Updated to:
```
{
       "code": "required_fields_missing",
       "message": "Required fields are missing: {}".format(", ".join(difference)),
}
```